src/app/services/schema/schema_processor_service.py

```diff
@@ -114,6 +115,32 @@ def calculate_next_schema_version(self, schema: dict):
             current_version_metadata, "sds_schema_version"
         )
 
+    def get_schema_version_from_properties(self, schema: dict) -> str:
+        """
+        Get schema version from the properties object of the schema.
+
+        Parameters:
+        schema (dict); schema being processed.
+        """
+
+        level_keys = ["properties", "schema_version", "const"]
+        return self.get_child_property(schema, level_keys)
+
+    def get_child_property(self, nested_dict: dict, keys: list):
+        """
+        Get a child property from a nested dictionary
+
+        Paramenters:
+        nested_dict (dict): A nested dictionary that is being fetched
+        keys (list): A list of keys within the nested dictionary that lead to the child property
+        """
+        field = nested_dict
+
+        for key in keys:
+            field = field[key]
```

Comment:
Better to have the result variable be called something different such as "result", as it avoids any issues of the return being the nested dictionary passed into the function in any side effect in the future.

Reply:
Ok will change